core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/io/SdkLengthAwareInputStream.java

```diff
@@ -48,8 +49,13 @@ public int read() throws IOException {
         }
 
         int read = super.read();
+
         if (read != -1) {
             remaining--;
+        } else if (remaining != 0) { // EOF, ensure we've read the number of expected bytes
+            throw new IllegalStateException("The request content has fewer bytes than the "
+                                            + "specified "
+                                            + "content-length: " + length + " bytes.");
```

Comment:
Was there a use-case that existed before, which we don't support anymore? E.g. someone specifies a content-length of 50 for a stream of length 100 so that they only send the first 50 bytes?

Reply:
This should still be supported. We only throw exception if content-length > number of bytes in the stream. If the stream contains more data, we truncate the stream (stop reading after content-length is reached.